ChefExtensionHandler/bin/chef-install.sh

```diff
@@ -59,10 +59,102 @@ install_chef_extension_gem(){
   fi
 }
 
-get_linux_distributor(){
-  lsb_release -i | awk '{print tolower($3)}'
+get_hostname (){
+  echo "Getting the hostname of this machine..."
+
+  host=`hostname -f 2>/dev/null`
+  if [ "$host" = "" ]; then
+    host=`hostname 2>/dev/null`
+    if [ "$host" = "" ]; then
+      host=$HOSTNAME
+    fi
+  fi
+
+  if [ "$host" = "" ]; then
+    echo "Unable to determine the hostname of your system!"
+    echo
+    echo "Please consult the documentation for your system. The files you need "
+    echo "to modify to do this vary between Linux distribution and version."
+    echo
+    exit 1
+  fi
+
+  echo "Found hostname: ${host}"
+}
+
+curl_check (){
+  echo "Checking for curl..."
+  if command -v curl > /dev/null; then
+    echo "Detected curl..."
+  else
+    echo "Installing curl..."
+    yum install -d0 -e0 -y curl
+  fi
+}
+
+install_from_local_package(){
+  # install chef
+  chef_client_installer=$(get_deb_installer)
+  installer_type="deb"
+  install_file $installer_type "$chef_client_installer"
+}
+
+install_from_repo(){
+  curl_check
+  get_hostname
+  yum_repo_path=/etc/yum.repos.d/chef_stable.repo
+  yum_repo_config_url="https://packagecloud.io/install/repositories/chef/stable/config_file.repo?os=el&dist=5&name=${host}"
```

Comment:
This is a common point of confusion for consumers of Chef outside the context of this extension, but the same RPM works on 5, 6, and 7 (it's built on 5 which enables that) and is tested on all three platforms by Chef.